superset/reports/notifications/webhook.py

```diff
@@ -0,0 +1,139 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import hmac
+import logging
+from typing import Dict, Optional
+
+import requests
+
+from superset import app
+from superset.reports.models import ReportRecipientType
+from superset.reports.notifications.base import BaseNotification
+from superset.reports.notifications.exceptions import NotificationError
+from superset.utils import json
+from superset.utils.decorators import statsd_gauge
+
+logger = logging.getLogger(__name__)
+
+
+class WebhookNotification(BaseNotification):  # pylint: disable=too-few-public-methods
+    """
+    Sends webhook notification to client host.
+    """
+
+    type = ReportRecipientType.WEBHOOK
+
+    @staticmethod
+    def _get_secret_token() -> str:
+        """
+        Retrieves the secret token for HMAC signature from the application configuration.
+        """
+        with app.app_context():
+            return app.config.get("WEBHOOK_SECRET")
+
+    def _get_host(self) -> str:
+        """
+        Retrieves the target host from the recipient configuration.
+        """
+        return json.loads(self._recipient.recipient_config_json)["target"]
+
+    def _generate_signature(
+        self, data: Dict[str, Optional[str | None | int]]
```

Comment:
we still support python 3.9, in order to use the new union type annotation we need to import:
```
from __future__ import annotations
```